File: Array/TypeScript/_1672_RichestCustomerWealth.ts
/**
 * Iteration
 * Time Complexity: BigO(N^2)
 * Space Complexity: BigO(1)
 */
function maximumWealth(accounts: number[][]): number {
  let highest = 0;
  for (let customer of accounts) {
    let sum = customer.reduce((acc, cur) => (acc += cur), 0);
    if (sum > highest) {
      highest = sum;
    }
  }
  return highest;
}

console.log(
  maximumWealth([
    [1, 2, 3],
    [3, 2, 1],
  ])
);
console.log(
  maximumWealth([
    [1, 5],
    [7, 3],
    [3, 5],
  ])
);
console.log(
  maximumWealth([
    [2, 8, 7],
    [7, 1, 3],
    [1, 9, 5],
  ])
);

File: Array/Python/_1672_RichestCustomerWealth.py
from typing import List


class Solution:
    # Iteration
    # Time Complexity: BigO(N^2)
    # Space Complexity: BigO(1)
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        highest = 0
        for customer in accounts:
            sum = 0
            for balance in customer:
                sum += balance
            if sum > highest:
                highest = sum
        return highest


ins = Solution

print(ins.maximumWealth(ins, [
    [1, 2, 3],
    [3, 2, 1],
]))
print(ins.maximumWealth(ins, [
    [1, 5],
    [7, 3],
    [3, 5],
]))
print(ins.maximumWealth(ins, [
    [2, 8, 7],
    [7, 1, 3],
    [1, 9, 5],
]))
